rcl/src/rcl/publisher.c

```diff
@@ -304,6 +306,40 @@ rcl_publisher_assert_liveliness(const rcl_publisher_t * publisher)
   return RCL_RET_OK;
 }
 
+rcl_ret_t
+rcl_publisher_wait_for_all_acked(const rcl_publisher_t * publisher, rcl_duration_value_t timeout)
+{
+  if (!rcl_publisher_is_valid(publisher)) {
+    return RCL_RET_PUBLISHER_INVALID;  // error already set
+  }
+
+  rmw_time_t rmw_timeout;
+  if (timeout > 0) {
+    rmw_timeout.sec = RCL_NS_TO_S(timeout);
+    rmw_timeout.nsec = timeout % 1000000000;
+  } else if (timeout < 0) {
+    rmw_time_t infinite = RMW_DURATION_INFINITE;
+    rmw_timeout = infinite;
+  } else {
+    rmw_time_t zero = RMW_DURATION_UNSPECIFIED;
+    rmw_timeout = zero;
+  }
+
+  rmw_ret_t ret = rmw_publisher_wait_for_all_acked(publisher->impl->rmw_handle, rmw_timeout);
+  if (ret != RMW_RET_OK) {
+    RCL_SET_ERROR_MSG(rmw_get_error_string().str);
```

Comment:
Should this be called when `ret == RMW_RET_TIMEOUT`?
I don't remember what we did in the rmw PRs, but `RMW_RET_TIMEOUT` is a non-error return code, so we shouldn't be setting any error neither calling RCL_SET_ERROR_MSG() here.

Reply:
Yes. Return `RMW_RET_TIMEOUT` isn't an error.